python_modules/dagster/dagster/_core/definitions/partition.py

```diff
@@ -1209,3 +1213,106 @@ def empty_subset(
         cls, partitions_def: Optional[PartitionsDefinition] = None
     ) -> "DefaultPartitionsSubset":
         return cls()
+
+
+class AllPartitionsSubset(
+    NamedTuple(
+        "_AllPartitionsSubset",
+        [
+            ("partitions_def", PartitionsDefinition),
+            ("instance_queryer", Optional["CachingInstanceQueryer"]),
```

Comment:
thoughts on making this arg non optional? to enforce that this subset is used within the span of a single request/iteration

Reply:
sure we can always relax this later i suppose